### PR TITLE
sveta.lua : max_query_size as variable + implement select ordered by last_request_time and by 2_week_count

### PR DESCRIPTION
**Pull request by grechkin-pogrebnyakov**

### CODE REVIEW

File: sveta.lua [hunk 1]
```diff
@@ -2,46 +2,144 @@
...
+    for i=19,7,-1 do
+        new_tuple[i] = new_tuple[i - 1]
+    end
+    new_tuple[6] = 0
+    return box.replace(space_no, new_tuple)
+end
+
+local function move_all_counters()
+    for t in box.space[space_no].index[0]:iterator(box.index.ALL) do
+        move_counters(t)
```
Comment (jirfag) [inline sveta.lua]: this loop can block tarantool for minutes, need to insert fiber.sleep on every nth iteration

File: sveta.lua [hunk 2]
```diff
@@ -2,46 +2,144 @@
...
+    end
+end
+
+local function move_all_counters_fiber()
+    while true do
+        local time = box.time()
+        local sleep_time = math.ceil(time/86400)*86400 - time + 1
+        box.fiber.sleep(sleep_time)
+        move_all_counters()
```
Comment (jirfag) [inline sveta.lua]: it's needed to print log message before moving counters and after moving, incuding total time of moving, it will help to debug issues.

Comment (jirfag) [inline sveta.lua]: Also if ```move_all_counters``` will raise error, loop will stop, therefore is should be ```pcall```-ed

File: sveta.lua [hunk 3]
```diff
@@ -2,46 +2,144 @@
...
+local function move_all_counters()
+    for t in box.space[space_no].index[0]:iterator(box.index.ALL) do
+        move_counters(t)
+    end
+end
+
+local function move_all_counters_fiber()
+    while true do
+        local time = box.time()
+        local sleep_time = math.ceil(time/86400)*86400 - time + 1
```
Comment (jirfag) [inline sveta.lua]: also print value of ```sleep_time```

File: sveta.lua [hunk 4]
```diff
@@ -2,46 +2,144 @@
...
-       end
-   end
+    uid = box.unpack('i', user_id)
+    while true do
+        -- delete by chuncks of dalete_chunk len
+        local tuples = {box.select_limit(space_no, 1, 0, delete_chunk, uid)}
+        if( table.getn(tuples) == 0 ) then
+            break
+        end
+        for _, tuple in ipairs(tuples) do
```
Comment (jirfag) [inline sveta.lua]: why not ```pairs```?

File: sveta.lua [hunk 5]
```diff
@@ -2,46 +2,144 @@
...
+-- space[4].index[1].key_field[0].fieldno = 0
+-- space[4].index[1].key_field[0].type = "NUM"
+-- space[4].index[1].key_field[1].fieldno = 3
+-- space[4].index[1].key_field[1].type = "NUM"
+-- space[4].index[2].type = "AVLTREE"
+-- space[4].index[2].unique = 0
+-- space[4].index[2].key_field[0].fieldno = 0
+-- space[4].index[2].key_field[0].type = "NUM"
+-- space[4].index[2].key_field[1].fieldno = 4
+-- space[4].index[2].key_field[1].type = "NUM"
```
Comment (jirfag) [inline sveta.lua]: it will be cool to see description of tuple structure here in comments, code is difficult to read without it

File: sveta.lua [hunk 6]
```diff
@@ -2,46 +2,166 @@
...
+        table.insert(resp, {tuple[index_of_user_id], tuple[index_of_query], tuple[index_of_total_counter], tuple[index_of_last_ts], tuple[index_of_2_week_counter]})
+        i = i + 1
+    end
+    return unpack(resp)
 end
+
+function select_recent(user_id, limit)
+    uid = box.unpack('i', user_id)
+    lim = box.unpack('i', limit)
+    return select_queries_by_index(uid, limit, 1)
```
Comment (Shmel1k) [inline sveta.lua]: Question: you get limit, then unpack it. Why do you use packed "limit" instead of unpacked "lim"?

**Pull request merged**